File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    temp = []
    for i in numbers:
        temp.append(str(i))
    temp.sort(key=lambda x: x*3, reverse=True)

    answer = ''
    for i in temp:
        answer += i
        
    return answer if answer[0] != '0' else '0'

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String[] s = new String[numbers.length];
        for(int i = 0; i < numbers.length; i++){
            s[i] = String.valueOf(numbers[i]);
        }
        Arrays.sort(s, (a,b) -> (b+a).compareTo(a+b) );
        StringBuilder sb = new StringBuilder();
        for(String k : s){
            sb.append(k);
        }
        if(sb.toString().charAt(0) == '0'){
            return "0";
        }
        
        return sb.toString();
    }
}